### DISCUSSION TITLE
Importing anything from pdfjs-dist results in very slow LSP exchanges

### DISCUSSION THREAD

**Discussion by angrybacon** [mozilla/pdf.js#18686] [Q&A]
Let me know if there exists a better place for this kind of issues

- `node` 22
- `pdfjs-dist` 4.6.82
- `typescript` 5.5.4
- `typescript-language-server` 4.3.3

I'm building a Next application using the above dependencies. The moment I comment my pdf.js usage `typescript-language-server` becomes usable again. The pdf.js usage is contained in this one file https://github.com/angrybacon/vntlnk/blob/master/modules/PoisonDartFrog/read.ts

How do I get to investigate further and see where exactly TypeScript is struggling? By unusable I mean that some LSP requests are very slow ie. 2-5 seconds and most end up timing out

**Comment by tamuratak** [comment]
I recommend adding `"skipLibCheck": true` to `tsconfig.json`. 

- https://www.typescriptlang.org/tsconfig/#skipLibCheck 

You can see `"skipLibCheck": true` is recommended in the reference.

I also recommend making `"include"` in `tsconfig.json` more specific. It would be better to avoid using `**/*.ts`.

- https://www.typescriptlang.org/tsconfig/#include

**Reply by angrybacon** [reply]
Missed that reply sorry. Thank you for the suggestions

It's not immediately clear but the `tsconfig` extends a common configuration file, where `skipLibCheck` is indeed `true`.

Can you give examples of what could go wrong with `**/*.ts`? As long as I'm excluding `node_modules` I can't think of a TypeScript files that I don't want to watch for 🤔 The Next.js application is using the `app` router and I'm not using a `src` folder so using such a loose pattern feels like a feature to me more than a danger

As for the slowness, I realized yesterday that my Vercel deployment was not supporting Node 22 yet and had to revert to Node 20 where I can't reproduce the slowness at all.
